doc/src/base/collections.md

```diff
@@ -236,7 +236,9 @@ Partially implemented by:
   * [`Iterators.Pairs`](@ref)
 
 ## Set-Like Collections
-A [`Set`](@ref) is a collection of elements with no duplicated elements and keys/indices. So in contrast to other types of collection, in a `Set` you can have only one of each element, and the order of elements isn't important (whereas an array can have multiple copies of an element and their order is remembered).
+A [`Set`](@ref) is a collection of elements with no duplicated elements and keys/indices.
+
+So in contrast to `Arrays`, in a `Set` you can have only one of each element, and the order of elements isn't important (whereas an array can have multiple copies of an element and their order is remembered). As a result, iterating over `Set` elements is slower than `Arrays`.
```

Comment:
The last sentence sounds like you were explaining some causality, which is IMO misleading. The performance depends on the specific implementation of a set, which for example could be a “plain” hash map (then iteration would most likely be a lot slower), some type of search tree (if the elements/hashes are ordered; then iteration would be about as fast as an array) or a hybrid structure equipped with an array of the unique elements (then iteration would be exactly as fast as ... an array). I would remove that last sentence.

Reply:
Also, I am generally not a big fan of single sentence paragraphs. So I would remove that whole commit.

Reply:
Current `Set` implementation is like this, so the sentence is true:

https://julialang.slack.com/archives/CF53T1DU4/p1580368042007300
> A `Set` stores the memory like a Dict, (i.e. a hash table), scattered all over the place, with empty spaces. It must jump around memory, often multiple times to fetch an element. And it must do a few simple checks every time it jumps

Reply:
I am not saying that you are wrong. My point is that if the documentation is based on an interface and not about a particular implementation, it should not state any properties that only depend on the implementation; at least not without referring to (and explaining) that implementation.

Reply:
The documentation is about the particular implementation here (not a general interface). If the implementation changes in the future, we need to update this.